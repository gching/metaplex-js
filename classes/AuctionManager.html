<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>AuctionManager | metaplex-js</title>
	<meta name="description" content="Documentation for metaplex-js">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="../assets/css/main.css">
	<script async src="../assets/js/search.js" id="search-script"></script>
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="../assets/js/search.json" data-base="..">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="../index.html" class="title">metaplex-js</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="../modules.html">metaplex-js</a>
				</li>
				<li>
					<a href="AuctionManager.html">AuctionManager</a>
				</li>
			</ul>
			<h1>Class AuctionManager</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel tsd-hierarchy">
				<h3>Hierarchy</h3>
				<ul class="tsd-hierarchy">
					<li>
						<span class="target">AuctionManager</span>
					</li>
				</ul>
			</section>
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Constructors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-constructor tsd-parent-kind-class"><a href="AuctionManager.html#constructor" class="tsd-kind-icon">constructor</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Properties</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="AuctionManager.html#acceptPayment" class="tsd-kind-icon">accept<wbr>Payment</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="AuctionManager.html#accountPublicKey" class="tsd-kind-icon">account<wbr>Public<wbr>Key</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="AuctionManager.html#auction" class="tsd-kind-icon">auction</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="AuctionManager.html#authority" class="tsd-kind-icon">authority</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="AuctionManager.html#key" class="tsd-kind-icon">key</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="AuctionManager.html#settings" class="tsd-kind-icon">settings</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="AuctionManager.html#state" class="tsd-kind-icon">state</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="AuctionManager.html#store" class="tsd-kind-icon">store</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="AuctionManager.html#vault" class="tsd-kind-icon">vault</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><a href="AuctionManager.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Methods</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><a href="AuctionManager.html#decode" class="tsd-kind-icon">decode</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Constructors</h2>
				<section class="tsd-panel tsd-member tsd-kind-constructor tsd-parent-kind-class">
					<a name="constructor" class="tsd-anchor"></a>
					<h3>constructor</h3>
					<ul class="tsd-signatures tsd-kind-constructor tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon">new <wbr>Auction<wbr>Manager<span class="tsd-signature-symbol">(</span>args<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Omit</span><span class="tsd-signature-symbol">&lt;</span><a href="AuctionManager.html" class="tsd-signature-type" data-tsd-kind="Class">AuctionManager</a><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">&quot;key&quot;</span><span class="tsd-signature-symbol">&gt;</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="AuctionManager.html" class="tsd-signature-type" data-tsd-kind="Class">AuctionManager</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L41">models/programs/metaplex/AuctionManager.ts:41</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>args: <span class="tsd-signature-type">Omit</span><span class="tsd-signature-symbol">&lt;</span><a href="AuctionManager.html" class="tsd-signature-type" data-tsd-kind="Class">AuctionManager</a><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">&quot;key&quot;</span><span class="tsd-signature-symbol">&gt;</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="AuctionManager.html" class="tsd-signature-type" data-tsd-kind="Class">AuctionManager</a></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Properties</h2>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="acceptPayment" class="tsd-anchor"></a>
					<h3>accept<wbr>Payment</h3>
					<div class="tsd-signature tsd-kind-icon">accept<wbr>Payment<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L37">models/programs/metaplex/AuctionManager.ts:37</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="accountPublicKey" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagOptional">Optional</span> account<wbr>Public<wbr>Key</h3>
					<div class="tsd-signature tsd-kind-icon">account<wbr>Public<wbr>Key<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L32">models/programs/metaplex/AuctionManager.ts:32</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="auction" class="tsd-anchor"></a>
					<h3>auction</h3>
					<div class="tsd-signature tsd-kind-icon">auction<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L35">models/programs/metaplex/AuctionManager.ts:35</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="authority" class="tsd-anchor"></a>
					<h3>authority</h3>
					<div class="tsd-signature tsd-kind-icon">authority<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L34">models/programs/metaplex/AuctionManager.ts:34</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="key" class="tsd-anchor"></a>
					<h3>key</h3>
					<div class="tsd-signature tsd-kind-icon">key<span class="tsd-signature-symbol">:</span> <a href="../enums/MetaplexKey.html" class="tsd-signature-type" data-tsd-kind="Enumeration">MetaplexKey</a><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L31">models/programs/metaplex/AuctionManager.ts:31</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="settings" class="tsd-anchor"></a>
					<h3>settings</h3>
					<div class="tsd-signature tsd-kind-icon">settings<span class="tsd-signature-symbol">:</span> <a href="AuctionManagerSettings.html" class="tsd-signature-type" data-tsd-kind="Class">AuctionManagerSettings</a></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L39">models/programs/metaplex/AuctionManager.ts:39</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="state" class="tsd-anchor"></a>
					<h3>state</h3>
					<div class="tsd-signature tsd-kind-icon">state<span class="tsd-signature-symbol">:</span> <a href="AuctionManagerState.html" class="tsd-signature-type" data-tsd-kind="Class">AuctionManagerState</a></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L38">models/programs/metaplex/AuctionManager.ts:38</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="store" class="tsd-anchor"></a>
					<h3>store</h3>
					<div class="tsd-signature tsd-kind-icon">store<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L33">models/programs/metaplex/AuctionManager.ts:33</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="vault" class="tsd-anchor"></a>
					<h3>vault</h3>
					<div class="tsd-signature tsd-kind-icon">vault<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L36">models/programs/metaplex/AuctionManager.ts:36</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class tsd-is-static">
					<a name="SCHEMA" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> SCHEMA</h3>
					<div class="tsd-signature tsd-kind-icon">SCHEMA<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">BorshSchema</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L10">models/programs/metaplex/AuctionManager.ts:10</a></li>
						</ul>
					</aside>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Methods</h2>
				<section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class tsd-is-static">
					<a name="decode" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> decode</h3>
					<ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class tsd-is-static">
						<li class="tsd-signature tsd-kind-icon">decode<span class="tsd-signature-symbol">(</span>accountWithPublicKey<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">AccountWithPublicKey</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="AuctionManager.html" class="tsd-signature-type" data-tsd-kind="Class">AuctionManager</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/metaplex/AuctionManager.ts#L45">models/programs/metaplex/AuctionManager.ts:45</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>accountWithPublicKey: <span class="tsd-signature-type">AccountWithPublicKey</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="AuctionManager.html" class="tsd-signature-type" data-tsd-kind="Class">AuctionManager</a></h4>
						</li>
					</ul>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class=" ">
						<a href="../modules.html">Exports</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
				<ul class="current">
					<li class="current tsd-kind-class">
						<a href="AuctionManager.html" class="tsd-kind-icon">Auction<wbr>Manager</a>
						<ul>
							<li class=" tsd-kind-constructor tsd-parent-kind-class">
								<a href="AuctionManager.html#constructor" class="tsd-kind-icon">constructor</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="AuctionManager.html#acceptPayment" class="tsd-kind-icon">accept<wbr>Payment</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="AuctionManager.html#accountPublicKey" class="tsd-kind-icon">account<wbr>Public<wbr>Key</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="AuctionManager.html#auction" class="tsd-kind-icon">auction</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="AuctionManager.html#authority" class="tsd-kind-icon">authority</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="AuctionManager.html#key" class="tsd-kind-icon">key</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="AuctionManager.html#settings" class="tsd-kind-icon">settings</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="AuctionManager.html#state" class="tsd-kind-icon">state</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="AuctionManager.html#store" class="tsd-kind-icon">store</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="AuctionManager.html#vault" class="tsd-kind-icon">vault</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class tsd-is-static">
								<a href="AuctionManager.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a>
							</li>
							<li class=" tsd-kind-method tsd-parent-kind-class tsd-is-static">
								<a href="AuctionManager.html#decode" class="tsd-kind-icon">decode</a>
							</li>
						</ul>
					</li>
				</ul>
				<ul class="after-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="../assets/js/main.js"></script>
</body>
</html>