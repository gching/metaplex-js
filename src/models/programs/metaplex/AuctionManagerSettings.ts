import { BorshSchema } from '../../utils/types';
import { WinningConfig } from './WinningConfig';
import { ParticipationConfig } from './ParticipationConfig';

export class AuctionManagerSettings {
  static SCHEMA: BorshSchema = new Map([
    [
      AuctionManagerSettings,
      {
        kind: 'struct',
        fields: [
          ['winningConfigs', [WinningConfig]],
          [
            'participationConfig',
            { kind: 'option', type: ParticipationConfig },
          ],
        ],
      },
    ],
    ...WinningConfig.SCHEMA,
    ...ParticipationConfig.SCHEMA,
  ]);

  winningConfigs: WinningConfig[] = [];
  participationConfig?: ParticipationConfig = undefined;

  constructor(args?: AuctionManagerSettings) {
    Object.assign(this, args);
  }
}
