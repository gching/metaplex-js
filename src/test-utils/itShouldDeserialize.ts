import { PublicKey } from '@solana/web3.js';
import { serialize } from 'borsh';
import { BorshSchema } from '../models/utils/types';

export function itShouldDeserialize<E>(
  cls: {
    SCHEMA: BorshSchema;
    // eslint-disable-next-line @typescript-eslint/ban-types
    decode: Function;
  },
  received: E,
  accountPubKey = PublicKey.default,
): void {
  const data = Buffer.from(serialize(cls.SCHEMA, received));

  const deserialized = cls.decode({
    pubkey: accountPubKey,
    account: {
      data,
    },
  }) as E;

  expect(deserialized).toDeepEqual(received);
}
