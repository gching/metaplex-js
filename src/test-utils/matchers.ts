import { isEqualWith } from 'lodash';
import BN from 'bn.js';

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace jest {
    interface Matchers<R> {
      toDeepEqual: <E = any>(expected: E) => R;
    }
  }
}

expect.extend({
  toDeepEqual<E = any>(received: E, expected: E) {
    const pass = isEqualWith(received, expected, (objValue, otherValue) => {
      if (BN.isBN(objValue) && BN.isBN(otherValue)) {
        return objValue.eq(otherValue);
      }
      return undefined;
    });

    if (pass) {
      return {
        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
        message: () => `expected ${received} not to deep equal ${expected}`,
        pass: true,
      };
    }

    return {
      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
      message: () => `expected ${received} to deep equal ${expected}`,
      pass: false,
    };
  },
});

export default undefined;
