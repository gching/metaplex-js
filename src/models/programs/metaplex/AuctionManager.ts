import { PublicKey } from '@solana/web3.js';
import { MetaplexKey } from './Store';
import { deserializeUnchecked } from 'borsh';
import { BorshSchema } from '../../utils/types';
import { AuctionManagerState } from './AuctionManagerState';
import { AuctionManagerSettings } from './AuctionManagerSettings';
import { AccountWithPublicKey } from '../../../utils/privateTypes';

export class AuctionManager {
  static SCHEMA: BorshSchema = new Map([
    [
      AuctionManager,
      {
        kind: 'struct',
        fields: [
          ['key', 'u8'],
          ['store', 'pubkey'],
          ['authority', 'pubkey'],
          ['auction', 'pubkey'],
          ['vault', 'pubkey'],
          ['acceptPayment', 'pubkey'],
          ['state', AuctionManagerState],
          ['settings', AuctionManagerSettings],
        ],
      },
    ],
    ...AuctionManagerState.SCHEMA,
    ...AuctionManagerSettings.SCHEMA,
  ]);

  key: MetaplexKey = MetaplexKey.AuctionManagerV1;
  accountPublicKey?: PublicKey;
  store: PublicKey;
  authority: PublicKey;
  auction: PublicKey;
  vault: PublicKey;
  acceptPayment: PublicKey;
  state: AuctionManagerState;
  settings: AuctionManagerSettings;

  constructor(args: Omit<AuctionManager, 'key'>) {
    Object.assign(this, args);
  }

  static decode(accountWithPublicKey: AccountWithPublicKey): AuctionManager {
    const auctionManager = deserializeUnchecked(
      AuctionManager.SCHEMA,
      AuctionManager,
      accountWithPublicKey.account.data,
    ) as AuctionManager;

    return new AuctionManager({
      ...auctionManager,
      accountPublicKey: accountWithPublicKey.pubkey,
    });
  }
}
