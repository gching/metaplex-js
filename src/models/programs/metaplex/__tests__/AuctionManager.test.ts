import { PublicKey } from '@solana/web3.js';
import { AuctionManager } from '../AuctionManager';
import { AuctionManagerSettings } from '../AuctionManagerSettings';
import {
  AuctionManagerState,
  AuctionManagerStatus,
} from '../AuctionManagerState';
import { itShouldDeserialize } from '../../../../test-utils/itShouldDeserialize';

describe('AuctionManager', () => {
  it('should instantiate with the correct values', () => {
    const auctionManager = new AuctionManager({
      auction: PublicKey.default,
      authority: PublicKey.default,
      accountPublicKey: PublicKey.default,
      store: PublicKey.default,
      vault: PublicKey.default,
      acceptPayment: PublicKey.default,
      state: new AuctionManagerState({
        winningConfigItemsValidated: 5,
        status: AuctionManagerStatus.Finished,
        winningConfigStates: [],
      }),
      settings: new AuctionManagerSettings({
        winningConfigs: [],
      }),
    });

    expect(auctionManager).toMatchSnapshot();
  });

  it('should deserialize with the correct values', () => {
    const auctionManager = new AuctionManager({
      auction: PublicKey.default,
      authority: PublicKey.default,
      accountPublicKey: PublicKey.default,
      store: PublicKey.default,
      vault: PublicKey.default,
      acceptPayment: PublicKey.default,
      state: new AuctionManagerState({
        winningConfigItemsValidated: 5,
        status: AuctionManagerStatus.Finished,
        winningConfigStates: [],
      }),
      settings: new AuctionManagerSettings({
        winningConfigs: [],
      }),
    });

    auctionManager.accountPublicKey = PublicKey.default;

    itShouldDeserialize(AuctionManager, auctionManager);
  });
});
