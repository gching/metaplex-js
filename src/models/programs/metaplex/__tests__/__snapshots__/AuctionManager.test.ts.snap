// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuctionManager should instantiate with the correct values 1`] = `
AuctionManager {
  "acceptPayment": PublicKey {
    "_bn": "00",
  },
  "accountPublicKey": PublicKey {
    "_bn": "00",
  },
  "auction": PublicKey {
    "_bn": "00",
  },
  "authority": PublicKey {
    "_bn": "00",
  },
  "key": 7,
  "settings": AuctionManagerSettings {
    "participationConfig": undefined,
    "winningConfigs": Array [],
  },
  "state": AuctionManagerState {
    "participationState": undefined,
    "status": 4,
    "winningConfigItemsValidated": 5,
    "winningConfigStates": Array [],
  },
  "store": PublicKey {
    "_bn": "00",
  },
  "vault": PublicKey {
    "_bn": "00",
  },
}
`;
